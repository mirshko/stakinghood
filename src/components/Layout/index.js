import React from "react";
import PropTypes from "prop-types";
import styled from "styled-components";

import "modern-normalize";
import "typeface-inter";
import "../../styles/global.css";

import Theme from "../Theme";

const Container = styled.div`
  display: grid;
  justify-items: center;
  min-height: 100vh;
`;

const Wrapper = styled.div`
  max-width: 400px;
  width: 100%;
`;

const Layout = ({ children }) => (
  <Theme>
    <Container>
      <Wrapper>{children}</Wrapper>
    </Container>
  </Theme>
);

Layout.propTypes = {
  children: PropTypes.node.isRequired
};

export default Layout;
