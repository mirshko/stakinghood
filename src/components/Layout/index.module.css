.container {
  display: grid;
  justify-items: center;
}

.wrapper {
  max-width: 480px;
  width: 100%;
}
