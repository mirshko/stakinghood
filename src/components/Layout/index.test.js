import React from "react";
import { shallow } from "enzyme";

import Layout from "./index";

describe("<Layout />", () => {
  it("it works", () => {
    const wrapper = shallow(
      <Layout>
        <p>Children</p>
      </Layout>
    );
    expect(wrapper).toMatchSnapshot();
  });
});
