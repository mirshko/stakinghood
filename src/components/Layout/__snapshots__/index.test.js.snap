// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Layout /> it works 1`] = `
<Theme>
  <styled.div>
    <styled.div>
      <p>
        Children
      </p>
    </styled.div>
  </styled.div>
</Theme>
`;
